src/eos-updater-fetch.c

```diff
@@ -113,27 +139,30 @@ content_fetch (GTask *task,
       goto error;
     }
 
-  message ("Fetch: %s:%s resolved to: %s", src, ref, sum);
+  message ("Fetch: %s:%s resolved to: %s", remote, ref, sum);
+  progress = ostree_async_progress_new_and_connect (update_progress, updater);
+
+  if (data->overridden_urls)
+    {
+      guint idx = g_random_int_range (0, g_strv_length (data->overridden_urls));
 
+      url_override = data->overridden_urls[idx];
```

Comment:
Hmm. Is choosing one randomly the best way to do it? I can't really think of any other off to top of my head but maybe we can be a bit smarter here. Perhaps leave a note in the code and we can get to it later as a low priority thing.